app/assets/javascripts/discourse/app/components/d-button.gjs

```diff
@@ -144,9 +141,19 @@ export default class DButton extends GlimmerComponentWithDeprecatedParentView {
     }
   }
 
+  get wrapperElement() {
+    const { href, type } = this.args;
+
+    return href
+      ? <template><a href={{href}} ...attributes>{{yield}}</a></template>
```

Comment:
Is it possible to use `@href`/`@type` directly in those templates?

Reply:
No, because `<template>` creates a template-only component, and using `@href` would refer to the arguments passed to that component (not DButton).

Similarly, referring to `this.` would access the template-only component instance, not the DButton instance.

So we need to make the values available in the local scope.